### PR TITLE
[PREVIEW] ROC-4294 Part admission CCJ

### PR DESCRIPTION
**Pull request by timja**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-4294

### Change description
A less hacky approach to part admit ccj than #845

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/main/common/components/ccj/paid-amount-summary.ts [hunk 1]
```diff
@@ -13,7 +13,8 @@ import { AbstractModelAccessor } from 'shared/components/model-accessor'
 /* tslint:disable:no-default-export */
 export abstract class AbstractPaidAmountSummaryPage<Draft> {
 
-  abstract createModelAccessor (): AbstractModelAccessor<Draft, PaidAmount>
+  abstract paidAmount (): AbstractModelAccessor<Draft, PaidAmount>
+  abstract amountSettledFor (claim: Claim, draft: Draft): number
```
Comment (ckalista) [inline src/main/common/components/ccj/paid-amount-summary.ts]: That's good 👍

File: src/main/features/claimant-response/helpers/amountHelper.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { DraftClaimantResponse } from 'claimant-response/draft/draftClaimantResponse'
+import { YesNoOption } from 'models/yesNoOption'
+import { Claim } from 'claims/models/claim'
+import { PartialAdmissionResponse } from 'claims/models/response/partialAdmissionResponse'
+import { ResponseType } from 'claims/models/response/responseType'
+
+export function calculateTotalAmount (claim: Claim, draft: DraftClaimantResponse): number | undefined {
```
Comment (ckalista) [inline src/main/features/claimant-response/helpers/amountHelper.ts]: As discussed, we could turn this into a class with static methods.

File: src/main/features/ccj/routes/paid-amount-summary.ts [hunk 3]
```diff
@@ -6,17 +6,22 @@ import { ccjPath, Paths } from 'features/ccj/paths'
...
     return new DefaultModelAccessor('paidAmount', () => new PaidAmount())
   }
 
   buildRedirectUri (req: express.Request, res: express.Response): string {
     const { externalId } = req.params
     return Paths.paymentOptionsPage.evaluateUri({ externalId: externalId })
   }
+
+  amountSettledFor (claim: Claim, draft: DraftCCJ): number {
+    return undefined
```
Comment (ckalista) [inline src/main/features/ccj/routes/paid-amount-summary.ts]: Is this as expected? For CCJ we don't have a settlement amount?

Comment (timja) [reply src/main/features/ccj/routes/paid-amount-summary.ts]: Yes, regular CCJ is a judgment by default for the full amount, whereas by admission you can settle for a lower amount

File: package.json [hunk 4]
```diff
@@ -110,7 +110,7 @@
     "codecov": "^3.0.4",
     "debug": "^3.0.0",
     "gulp-livereload": "^3.8.0",
-    "gulp-nodemon": "^2.0.2",
+    "gulp-nodemon": "^2.2.1",
```
Comment (timja) [inline package.json]: why this change?

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cmc-citizen-frontend",
-  "version": "1.1.1",
+  "version": "1.9.4",
```
Comment (timja) [inline package.json]: why this change?

File: src/main/common/components/ccj/paid-amount-summary.njk [hunk 6]
```diff
@@ -8,9 +8,8 @@
   <div class="column-two-thirds">
 
     <div class="form-group">
-      <p>
-        {{ t('Judgment will be entered against the defendant for this amount.') }}
```
Comment (robertparkinson) [inline src/main/common/components/ccj/paid-amount-summary.njk]: Why is this being removed? It is in the prototype: 
https://hmcts-cmc-prototype.herokuapp.com/prototypes/features-to-be-built/dashboard/applications-and-judgments/default-judgment/claim-amount?radio-indent-group2=Yes&num2=500

File: src/main/common/components/ccj/paid-amount.njk [hunk 7]
```diff
@@ -22,9 +22,7 @@
 
           <div class="panel panel-border-narrow js-hidden" id="option-yes">
 
-            <p>{{ t('Total claim amount: {{ totalAmount }}', { totalAmount: totalAmount | numeral }) }}</p>
-
-            <input type="hidden" name="claimedAmount" value="{{totalAmount}}" />
+            <input type="hidden" name="claimedAmount" value="{{ totalAmount }}" />
```
Comment (robertparkinson) [inline src/main/common/components/ccj/paid-amount.njk]: Why do we have a hidden input on this page?

File: src/main/common/components/ccj/paid-amount-summary.njk [hunk 8]
```diff
@@ -20,13 +19,22 @@
...
+        {% if not amountSettledFor %}
+          <tr>
+            <th scope="row">{{ t('Claim amount') }}</th>
+            <td class="numeric last">
+              <span class="font-small">{{ claim.claimData.amount.totalAmount() | numeral }}</span>
+            </td>
+          </tr>
+        {% else %}
+          <tr>
+            <th scope="row">{{ t('Amount you are settling for') }}</th>
```
Comment (robertparkinson) [inline src/main/common/components/ccj/paid-amount-summary.njk]: I'm pretty sure this should say "Amount you're settling for" but struggling to find the prototype page for this content

File: src/main/common/components/ccj/paid-amount.njk (hunk 7 above)
Comment (timja) [reply src/main/common/components/ccj/paid-amount.njk]: validation, you can't do more than total amount

File: src/main/common/components/ccj/paid-amount-summary.njk (hunk 6 above)
Comment (timja) [reply src/main/common/components/ccj/paid-amount-summary.njk]: Marc asked for it to be removed

File: src/main/common/components/ccj/paid-amount-summary.njk (hunk 8 above)
Comment (timja) [reply src/main/common/components/ccj/paid-amount-summary.njk]: There is no prototype, this content was written by Chris and signed off by marc

**Comment by timja** [comment]
Merging this in, can do more iteration on content later if required

File: src/main/features/claimant-response/helpers/amountHelper.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+import { Claim } from 'claims/models/claim'
+import { PartialAdmissionResponse } from 'claims/models/response/partialAdmissionResponse'
+import { ResponseType } from 'claims/models/response/responseType'
+
+export class AmountHelper {
+  static calculateTotalAmount (claim: Claim, draft: DraftClaimantResponse): number {
+    const settledAmount = AmountHelper.calculateAmountSettledFor(claim, draft)
+
+    if (settledAmount) {
+      return settledAmount + claim.claimData.feeAmountInPennies / 100
```
Comment (sabahirfan) [inline src/main/features/claimant-response/helpers/amountHelper.ts]: please check whether we need to add interest here as well?
